src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -170,6 +171,8 @@ public WebApplication Build()
             ((IConfigurationBuilder)Configuration).Sources.Clear();
             Configuration.AddConfiguration(_builtApplication.Configuration);
 
+            _builtApplication.Properties[WebApplicationBuilderKey] = true;
```

Comment:
What's the purpose of this property? Trying to grok where it is used...

Reply:
Nevermind! I see you are using this in the extension methods to check if they are invoked on a WebApplicationBuilder. Correct?

Reply:
Yes, I added this "hack" to clearly indicate that the builder is a WebApplicationBuilder, where we make use of "creative" pipeline construction techniques 😄